ecosystem/typescript/sdk/examples/typescript-esm/multi_ed25519_to_multisig.ts

```diff
@@ -0,0 +1,437 @@
+  const bitmap = createBitmapFromDiff(accountAddresses, signingAddresses);
+
+  // Step 5.
+  // Assemble a MultiEd25519 signed proof struct with the gathered signatures.
+  // This represents the multisig signed struct by all owners. This is used as proof of authentication by the overall MultiEd25519 account, since there's no signer
+  // checked in the entry function we're using.
+  const multiSigStruct = createMultiSigStructFromSignedStructs(structSignatures, bitmap);
+
+  // Create test metadata for the multisig account post-creation
+  const metadataKeys = ["key 123", "key 456", "key 789"];
+  const metadataValues = [new Uint8Array([1, 2, 3]), new Uint8Array([4, 5, 6]), new Uint8Array([7, 8, 9])];
+
+  // Pack the signed multi-sig struct into the entry function payload with the number of signatures required + multisig account info
+  const entryFunctionPayload = createWithExistingAccountAndRevokeAuthKeyPayload(
+    multiSigAddress,
+    multiSigPublicKey,
+    multiSigStruct,
+    accountAddresses,
+    NUM_SIGNATURES_REQUIRED,
+    metadataKeys,
+    metadataValues,
+  );
+
+  // Step 6.
+  // Call the `0x1::multisig_account::create_with_existing_account_and_revoke_auth_key` function
+  // At this point, there are two options to submit the transaction:
+  //   Since you've already authenticated the signed struct message with the required K of N accounts, you do not need to construct a MultiEd25519 authenticated signature.
+  //   You can submit the transaction as literally anyone, because the entry function does not check the sender. The transaction is validated with the multisig signed struct.
+  //   This is a much simpler flow, since it does not require that each of the owners in the original MultiEd25519 account sign the transaction.
+  const OPTION_ONE = false;
+  if (OPTION_ONE) {
+    //////////////////////////////////////////////////////////////////////////////////////////
+    //           Option 1: Submitting the transaction as a single Ed25519 account           //
+    //////////////////////////////////////////////////////////////////////////////////////////
+    const randomAccount = new AptosAccount();
+    await faucetClient.fundAccount(randomAccount.address(), 100_000_000);
+
+    // The sender here is essentially just paying for the gas fees.
+    const txn = await provider.generateSignSubmitTransaction(randomAccount, entryFunctionPayload, {
+      expireTimestamp: BigInt(Math.floor(Date.now() / 1000) + 60),
+    });
+
+    // Step 7.
+    // Wait for the transaction to complete, then observe and assert that the authentication key for the original MultiEd25519 account
+    // has been rotated and all capabilities revoked.
+    const txnInfo = await provider.waitForTransactionWithResult(txn);
+    printRelevantTxInfo(txnInfo as Types.UserTransaction);
+    assertChangesAndPrint(provider, multiSigAddress, sequenceNumber, false, metadataKeys, metadataValues);
+  } else {
```

Comment:
Is this option 2 really useful? Once the challenge message has been signed by enough signatures to form the multied25519 key, it's better submitted by a single account? The only use case I ever see here is to not have to pay gas, but gas is so low, it doesn't really matter. Probably costs cents to do the migration

Reply:
No, it's not really, haha. It was more just to show how to sign a transaction that needs a proof with a multied account.

Lowkey I didn't realize until the very end that the auth key revocation function in multisig_account.move doesn't need a `&signer`, but I was so far into it I just finished the example as if it needed it